script.js

```diff
@@ -0,0 +1,364 @@
+      // Handle the leading zero for the first number
+      if (first.value.startsWith("0") && first.value.slice(1, 2) !== ".") {
+        first.value = first.value.substring(1);
+      }
+      // Allow a new operation thanks to a previous result
+      if (allowNewNumber) {
+        first.value = operandEvent;
+        result = "";
+        allowNewNumber = false;
+      }
+    }
+    // Save the first digit for the first number
+    if (first.unsaved) {
+      first.value = operandEvent;
+      saveData(first);
+    }
+  }
+
+  function addOperator(operatorEvent) {
+    // Evaluate expression when chaining operations with a new sign
+    if (first.saved && operator.saved && second.saved) {
+      evaluateExpression(first, operator, second);
+    }
+    if (operatorEvent === "-" && operator.saved && second.unsaved) {
+      // Allow a leading minus sign for the second number
+      second.value = operatorEvent;
+      saveData(second);
+      // Handle both negative sign and number
+      if (operator.value === "-" && second.value.startsWith("-")) {
+        operator.value = "+";
+        resetData(second);
+      }
+      // Handle a plus sign and followed by a negative second number
+      if (operator.value === "+" && second.value.startsWith("-")) {
+        operator.value = "-";
+        second.value = second.value.substring(1);
+        resetData(second);
+      }
+    }
+    // Save the sign for the expression
+    if (first.saved && operator.unsaved && first.value !== "-") {
+      operator.value = operatorEvent;
+      saveData(operator);
+    }
+    // Allow a leading minus sign for the first number
+    if (operatorEvent === "-" && first.unsaved && operator.unsaved) {
+      first.value = operatorEvent;
+      saveData(first);
+    }
+  }
+
+  function evaluateExpression(first, operator, second) {
+    // Handle a complete expression
+    if (first.saved && operator.saved && second.saved) {
+      // Handle a single minus sign as a second number, like 25x-
+      if (second.value === "-") {
+        return;
+      }
+      result = operate(first.value, operator.value, second.value);
+    }
+    // Handle a single first number followed by a sign
+    if (first.saved && operator.saved && second.unsaved) {
+      second.value = first.value;
+      result = operate(first.value, operator.value, second.value);
+    }
+    // Handle a single first number
+    if (first.saved && operator.unsaved && second.unsaved) {
+      result = first.value;
+    }
+    // Handle an empty expression
+    if (first.unsaved && operator.unsaved && second.unsaved) {
+      result = "0";
+    }
+    // Allow to chain expression when adding a sign
+    if (result !== "Error") {
+      first.value = result;
+      allowNewNumber = true;
+      [operator, second].forEach(element => resetData(element));
+    } else {
+      [first, operator, second].forEach(element => resetData(element));
+    }
+  }
+
+  function operate(first, sign, second) {
```

Comment:
`operate` is not a very good name, I'd call it `calculate`. You don't need to return String for all these results, it'd be better for the caller to do it.

Reply:
To be consistent and clearer, could we call `sign` `operator`?